hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -145,7 +182,74 @@ private static AutoField getSourceField(Query query) {
                 throw new UncheckedIOException(e);
             }
         } else {
-            return new NoopField();
+            return null;
+        }
+    }
+
+    private interface FrameReadTaskSourceProvider {
```

Comment:
this seems like it is just the jdk interface `Supplier`. eg. `Supplier<List<QueryTaskSource>>`. Even the method signatures are the same.